packages/components/src/DateTimePickers/DateTimePicker/DateTimePicker.component.js

```diff
@@ -0,0 +1,160 @@
+		this.onSelectCalendarMonth = this.onSelectCalendarMonth.bind(this);
+		this.onSelectCalendarYear = this.onSelectCalendarYear.bind(this);
+		this.onSelectCalendarMonthYear = this.onSelectCalendarMonthYear.bind(this);
+		this.onSelectDate = this.onSelectDate.bind(this);
+		this.onSelectTime = this.onSelectTime.bind(this);
+	}
+
+	componentWillReceiveProps(nextProps) {
+		const propsUpdated = this.props.selection !== nextProps.selection;
+
+		if (!propsUpdated) {
+			return;
+		}
+
+		const newSelectedDate = nextProps.selection && nextProps.selection.date;
+		const newSelectedTime = nextProps.selection && nextProps.selection.time;
+		const needToUpdateDate = newSelectedDate !== this.state.selectedDate;
+		const needToUpdateTime = newSelectedTime !== this.state.selectedTime;
+		const needToUpdateState = needToUpdateDate || needToUpdateTime;
+
+		if (!needToUpdateState) {
+			return;
+		}
+
+		const needToUpdateCalendar = needToUpdateDate && newSelectedDate !== undefined;
+
+		const newState = {
+			...(needToUpdateDate && { selectedDate: newSelectedDate }),
+			...(needToUpdateTime && { selectedTime: newSelectedTime }),
+			...(needToUpdateCalendar && {
+				calendar: {
+					monthIndex: getMonth(newSelectedDate),
+					year: getYear(newSelectedDate),
+				},
+			}),
+		};
+
+		this.setState(newState);
+	}
+
+	onSelectDate(selectedDate) {
+		this.setState({ selectedDate }, () => {
+			this.trySubmit();
+		});
+	}
+
+	onSelectTime(selectedTime) {
+		this.setState({ selectedTime }, () => {
+			this.trySubmit();
+		});
+	}
+
+	onSelectCalendarMonthYear(newCalendar) {
+		this.setState(previousState => ({
+			calendar: {
+				...previousState.calendar,
+				...newCalendar,
+			},
+		}));
+	}
+
+	onSelectCalendarMonth(monthIndex) {
+		this.onSelectCalendarMonthYear({ monthIndex });
+	}
+
+	onSelectCalendarYear(year) {
+		this.onSelectCalendarMonthYear({ year });
+	}
+
+	setView(isDateTimeView) {
+		this.setState({ isDateTimeView });
+	}
+
+	trySubmit() {
+		if (this.state.selectedDate !== undefined &&
+			this.state.selectedTime !== undefined
+		) {
+			this.props.onSubmit({
+				date: this.state.selectedDate,
+				time: this.state.selectedTime,
+			});
+		}
+	}
+
+	render() {
+		let viewElement;
+
+		if (this.state.isDateTimeView) {
+			viewElement = (<DateTimeView
+				onClickTitle={this.setMonthYearView}
+				calendar={this.state.calendar}
+				onSelectMonthYear={this.onSelectCalendarMonthYear}
+				onSelectDate={this.onSelectDate}
+				selectedDate={this.state.selectedDate}
+				onSelectTime={this.onSelectTime}
+				selectedTime={this.state.selectedTime}
+			/>);
+		} else {
+			viewElement = (<MonthYearView
+				onClickBack={this.setDateTimeView}
+				selectedMonthIndex={this.state.calendar.monthIndex}
+				selectedYear={this.state.calendar.year}
+				onSelectMonth={this.onSelectCalendarMonth}
+				onSelectYear={this.onSelectCalendarYear}
+			/>);
+		}
+
+		return (
+			<div className={theme.container}>
+				{viewElement}
+			</div>
+		);
```

Comment:
How to you achieve it ? With duplication of the `div` container inside each `if/else` block ? ?

Reply:
Yes. It's not an issue to duplicate one line of JSX for comprehension.

Reply:
Having two lines that needs to be identical is a code synchronization maintenance issue.
I don't see any reason why it's actually complex or not understandable.
=> We decide what view is active and how it's define and then use it in the render. Here there is 2 views but it could have been more.